sql/src/main/java/io/crate/planner/node/management/ExplainPlan.java

```diff
@@ -63,15 +83,49 @@ public void execute(DependencyCarrier executor,
                         RowConsumer consumer,
                         Row params,
                         Map<SelectSymbol, Object> valuesBySubQuery) {
-        Map<String, Object> map;
 
         if (context.enabled()) {
-            ProfilingContext.TimerToken timerToken = context.startTiming("Execute");
-            subPlan.execute(executor, plannerContext, NOOP_ROW_CONSUMER, params, valuesBySubQuery);
-            context.stopTiming(timerToken);
-            map = context.build();
+            assert subPlan instanceof LogicalPlan : "subPlan must be a LogicalPlan";
+            LogicalPlan plan = (LogicalPlan) subPlan;
+            /**
+             * EXPLAIN ANALYZE does not support analyzing {@link io.crate.planner.MultiPhasePlan}s
+             */
+            if (plan.dependencies().isEmpty()) {
```

Comment:
could we split up this up a bit? maybe by extracting the `whenComplete` logic into a named function